### DISCUSSION TITLE
Client <unknown> - MQTT to Home Assistant not working

### DISCUSSION THREAD

**Discussion by cofade** [jomjol/AI-on-the-edge-device#1151] [Q&A]
I have now tried literally everything I found on the internet to get my data via MQTT into Home Assistant, but it seems to be impossible... Nevertheless, I have been able to narrow down where the error could originate from.

1. When installing a mosquitto broker on my Windows PC and setting the watermeter settings accordingly, the broker receives its MQTT messages.
2. When publishing MQTT messages from my Windows PC to my Home Assistant RPi, they are received without any issue.
3. But: When publishing from the watermeter to Home Assistant, it gives the following error message:

> 2022-10-09 01:24:54: New connection from 192.168.178.51:62450 on port 1883.
> 2022-10-09 01:24:54: Sending CONNACK to 192.168.178.51 (0, 5)
> 2022-10-09 01:24:54: Client <unknown> disconnected, not authorised.

From the watermeter's log I get this output:

> 2022-10-09T01:22:16: ClassFlowCNNGeneral::getReadout#result= 00517
> 2022-10-09T01:22:16: 	FlowControll.doFlow - ClassFlowMQTT
> 2022-10-09T01:22:16: MQTT - Failed to publish 'watermeter/connection'!
> 2022-10-09T01:22:16: MQTT - Re-running init...!
> 2022-10-09T01:22:16: MQTT - Init
> 2022-10-09T01:22:18: MQTT - Init successful
> 2022-10-09T01:22:18: MQTT - Failed to publish 'watermeter/connection'!
> 2022-10-09T01:22:18: task_autodoFlow - round done

I found on the internet that connectivity issues also with tasmota devices with this error messages occurred after the recent Mosquitto update in Home Assistant, which prevented anonymous logins.

So now my question is: Why is the watermeter trying to use a anonymous MQTT login despite the full definition of all necessary credentials?
![image](https://user-images.githubusercontent.com/53254078/194731165-38b08d46-dd63-49e2-8988-7cf0668b2a6d.png)

**Comment by caco3** [comment]
There are some issues with MQTT as of now, for some it works very stable, others have issues.
I myself see that it sporadically fails to publish. Then it re-inits and afterwards publishing successes.
MQTT stability is something we need to work on.

However:
Your issues seem not related to AI-on-the-edge! Looking on your point 1, shows that the messages get sent (in opposite to what your log sais).

On my setup, I have 2 things different:
1. I have a separate docker container running the broker
2. I don't use login credentials

Maybe you could also try the later as a test?
Since Tasmota also has issues, it might more be an issue of the broker within HA.
